Add users, user authentication

mileszs opened the pull request:
I used [this](http://nithinbekal.com/posts/phoenix-authentication/) as a reference. Its first comment was from @stevegrossi, which I thought was pretty grand.

I left off registration, as I don't want users to register for the site. The end goal is to have admins that we create manually, and job posters who can only "login" via the token-ized URLs in their inboxes.

I was able to load up a console (`iex -S phoenix.server`) and run the following to create a user:

``` elixir
changeset = ElixirJobBoard.User.changeset(%ElixirJobBoard.User{}, %{ "email" => "miles@mileszs.com", "password" => "password")

ElixirJobBoard.Registration.create(changeset, ElixirJobBoard.Repo)
```

Then, I could log in using those credentials. Whee!

PS - I know, we still haven't written any tests. In this experimental phase, I'm okay with that. We should totally write tests before this thing goes live, though. :-) We should probably write tests before we start polishing / writing complicated features. I've already looked at [a code coverage tool](http://tech.findmypast.com/elixir-maintainability/#excoveralls) and [a code linter](http://tech.findmypast.com/elixir-maintainability/#excoveralls).

mileszs committed:
Add users, user authentication

mileszs committed:
Add users, user authentication

mileszs committed:
Add seeds for user accounts

mileszs committed:
Add seeds for user accounts

mileszs commented:
Ended up adding some seeds for user accounts to play with in dev. One of those was added just for you, @andrewariley87. I hope you appreciate it.

andrewariley87 commented:
#### Should this be a concern with using comeonin

![screen shot 2016-09-26 at 11 39 40 pm](https://cloud.githubusercontent.com/assets/12799112/18859506/c01c8030-8442-11e6-998d-5f765b197e8f.png)

andrewariley87 commented on priv/repo/seeds.exs:
```diff
@@ -9,3 +9,25 @@
+)
+zeisloft_changeset = ElixirJobBoard.User.changeset(
+  %ElixirJobBoard.User{},
+  %{ "email" => "nick.zeisloft.iu@pacers.com", "password" => "password" }
```
smh

andrewariley87 commented:
👍  with comments

mileszs committed:
Update comeonin encryption lib

mileszs committed:
Update comeonin encryption lib

mileszs committed:
Actually use the default_password variable

mileszs committed:
Actually use the default_password variable

mileszs commented:
@andrewariley87 Updated comeonin. Does it resolve your warning?

andrewariley87 commented:
@mileszs Yes the update got rid of the error but one of the tests is failing.
![screen shot 2016-09-27 at 10 04 54 pm](https://cloud.githubusercontent.com/assets/12799112/18898274/7f3bae26-84fe-11e6-8d97-fec51428133d.png)

andrewariley87 committed:
Fixes failing test

Fixes "changeset with valid attributes" test in user_test.exs.

andrewariley87 committed:
Fixes failing test

Fixes "changeset with valid attributes" test in user_test.exs.

andrewariley87 commented:
@mileszs I added a commit to fix the failing test.

I am ready to merge this in if you are.

andrewariley87 committed:
Posts Controller Tests

Adds tests for the Posts controller.
Also adds new mix called ExMachina that allow you to create factories
for your Ecto Models because Phoenix does not have fixtures.
Fixes an unsuccessful create and update actions because they were
not working.

mileszs committed:
Add test for registration

mileszs committed:
Add test for registration

mileszs commented:
I added a test for registration. What do you think, @andrewariley87 ?

mileszs committed:
Make thest a bit more robust (I think)

mileszs committed:
Make thest a bit more robust (I think)

andrewariley87 commented:
@mileszs test looks good, Do you want me to merge this PR in?

mileszs committed:
Merge pull request #5 from indyhackers/add_controller_tests

Posts Controller Tests

mileszs commented:
@andrewariley87 Yes, that'd be great. Let me rebase, here.

mileszs committed:
Merge branch 'add-user-authentication' of github.com:indyhackers/elixir-job-board into add-user-authentication
